src/main/java/jsr223/docker/compose/DockerComposeScriptEngine.java

```diff
@@ -96,41 +90,37 @@ public Object eval(String script, ScriptContext context) throws ScriptException
                     context.getReader());
 
             // Wait for process to exit
-            int returnValue = process.waitFor();
-
-            // Stop containers
-            SingletonProcessBuilderFactory
-                    .getInstance()
-                    .getProcessBuilder(DockerComposeCommandCreator.createDockerComposeStopCommand())
-                    .start().waitFor();
-
-            // Remove containers
-            SingletonProcessBuilderFactory
-                    .getInstance()
-                    .getProcessBuilder(DockerComposeCommandCreator.createDockerComposeRemoveCommand())
-                    .start().waitFor();
-
-            return returnValue;
-
+            return process.waitFor();
         } catch (IOException e) {
             log.warn("Failed to execute Docker Compose.", e);
         } catch (InterruptedException e) {
-            // Thread was interrupted somehow, now we need to make sure that
-            // the container are stopped even though several interrupts could
-            // occur. Therefore start a thread which will remove all container.
-            log.warn("Execution was terminated and container might need to be terminated manually.");
+            log.info("Container execution interrupted. " + e.getMessage());
         } finally {
+            try {
+                // Reset thread's interrupt flag
+                Thread.interrupted();
+                stopAndRemoveContainers().waitFor();
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1148)